### PR TITLE
Token Auth Backend: Token request function didn`t set Authorization Header

### PR DESCRIPTION
**Pull request by mariusbertram**
In the Token Auth Backend the functin request_token were missing the Authorization Header which failed in not receiving a Bearer Token.

### CODE REVIEW

File: oras/auth/token.py [hunk 1]
```diff
@@ -66,6 +65,7 @@ def authenticate_request(
             return headers, False
 
         # If we have a token, set auth header (base64 encoded user/pass)
+        # Else if
```
Comment (vsoch) [inline oras/auth/token.py:68]: Did you mean to write another comment here?

File: oras/auth/token.py [hunk 2]
```diff
@@ -156,6 +160,7 @@ def request_anonymous_token(self, h: auth_utils.authHeader) -> bool:
         # From https://docs.docker.com/registry/spec/auth/token/ section
         # We can get token OR access_token OR both (when both they are identical)
         data = response.json()
+        print("data: %s" % data)
```
Comment (vsoch) [inline oras/auth/token.py:163]: This should probably be debug or removed for the PR.

**Review by vsoch — CHANGES_REQUESTED** [review]

File: oras/auth/token.py (hunk 2 above)
Comment (mariusbertram) [reply oras/auth/token.py:163]: Yeah dubbing

File: oras/auth/token.py (hunk 1 above)
Comment (mariusbertram) [reply oras/auth/token.py]: I tried to modify this function, but in the end i didn't

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,7 @@ and **Merged pull requests**. Critical items to know are:
 The versions coincide with releases on pip. Only major versions will be released as tags on Github.
 
 ## [0.0.x](https://github.com/oras-project/oras-py/tree/main) (0.0.x)
+ - add missing basic auth data for request token function in token auth backend (0.2.2)
```
Comment (vsoch) [inline CHANGELOG.md:17]: Just bump the oras/version.py and this will be ready to go!

Comment (mariusbertram) [reply CHANGELOG.md:17]: Done

**Review by vsoch — APPROVED** [review]

**Pull request merged**